binlog_streamer.go

```diff
@@ -114,9 +114,10 @@ func (s *BinlogStreamer) Run() {
 		var ev *replication.BinlogEvent
 		var timedOut bool
 
-		err := WithRetries(5, 0, s.logger, "get binlog event", func() (er error) {
-			ctx, _ := context.WithTimeout(context.Background(), 500*time.Millisecond)
-			ev, er = s.binlogStreamer.GetEvent(ctx)
+		err := WithRetriesContext(ctx, 5, 0, s.logger, "get binlog event", func() (er error) {
+			eventCtx, eventCancel := context.WithTimeout(ctx, 500*time.Millisecond)
+			defer eventCancel()
+			ev, er = s.binlogStreamer.GetEvent(eventCtx)
```

Comment:
nit: can we rename `er` to `err` and just define it as a `var` underneath `timeout`?